processor/src/fast/u32_ops.rs

```diff
@@ -33,15 +35,21 @@ impl FastProcessor {
             let b = self.stack_get(1).as_int();
             let a = self.stack_get(2).as_int();
 
-            // Check that a, b, and c are u32 values.
+            // Collect all invalid u32 values before returning an error
+            let mut invalid_values = Vec::new();
+
             if a > u32::MAX as u64 {
-                return Err(ExecutionError::not_u32_value(Felt::new(a), ZERO, err_ctx));
+                invalid_values.push(Felt::new(a));
             }
             if b > u32::MAX as u64 {
-                return Err(ExecutionError::not_u32_value(Felt::new(b), ZERO, err_ctx));
+                invalid_values.push(Felt::new(b));
             }
             if c > u32::MAX as u64 {
-                return Err(ExecutionError::not_u32_value(Felt::new(c), ZERO, err_ctx));
+                invalid_values.push(Felt::new(c));
+            }
+
+            if !invalid_values.is_empty() {
+                return Err(ExecutionError::not_u32_values(invalid_values, ZERO, err_ctx));
             }
```

Comment:
Question: if all values are valid, this shouldn't result in vector allocation, right? Also, if we wanted to optimize it even more for the happy path, we could probably do something like this:
```rust
if a > u32::MAX as u64 || b > u32::MAX as u64 || c > u32::MAX as u64 {
    // TODO: build the error here
}
```